### PR TITLE
feat (docs): add genui sdk ui page

### PR DESCRIPTION
**Pull request by nicoalbanese**

### CODE REVIEW

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+description: Learn how to build Generative UI with AI SDK UI.
+---
+
+# Generative UI
+
+Generative UI is the process of allowing a large language model (LLM) to go beyond text and "generate UI". This creates a more engaging and AI-native experience for users.
+
+<WeatherSearch />
+
+At the core of generative UI are [ tools ](/docs/ai-sdk-core/tools-and-tool-calling), which are like programs you provide to the model to perform specialized tasks like getting the weather in a location. The model can decide when and how to use these tools based on the context of the conversation.
```
Comment (lgrammel) [inline content/docs/04-ai-sdk-ui/04-generative-ui.mdx:12]: `which are functions that you provide` (audience are developers)

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+Generative UI is simply the process of piping the results of a tool call to a React component. Here's how the process works:
```
Comment (lgrammel) [inline content/docs/04-ai-sdk-ui/04-generative-ui.mdx:14]: cut `simply`

Comment (lgrammel) [reply content/docs/04-ai-sdk-ui/04-generative-ui.mdx:14]: `Here's how it works:`

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+1. You provide the model with a prompt or conversation history, along with a set of tools.
+2. Based on the context, the model may decide to call a tool.
+3. If a tool is called, it will execute and return data.
```
Comment (lgrammel) [inline content/docs/04-ai-sdk-ui/04-generative-ui.mdx:18]: There is also the GenUI path that's using tools w/o `execute` (might come later)

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+4. This data can then be piped to a React component for rendering.
+5. If no relevant tool is available or a tool isn't necessary (ie. summarising actions), the model will generate text.
```
Comment (lgrammel) [inline content/docs/04-ai-sdk-ui/04-generative-ui.mdx:20]: The model now always generates text at the end?

**Review by lgrammel — APPROVED** [review]

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx [hunk 5]
```diff
@@ -0,0 +1,347 @@
+---
+title: Generative UI
+description: Learn how to build Generative UI with AI SDK UI.
+---
+
+# Generative UI
```
Comment (jeremyphilemon) [inline content/docs/04-ai-sdk-ui/04-generative-ui.mdx:6]: I would expand the title as "Generative User Interfaces"

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx (hunk 2 above)
Comment (jeremyphilemon) [reply content/docs/04-ai-sdk-ui/04-generative-ui.mdx:14]: I would rephrase to avoid the word "piping" since it's not immediately obvious what it means

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+import { streamText, convertToCoreMessages } from 'ai';
+import { tools } from '@/ai/tools';
+
+export async function POST(request: Request) {
+  const { messages } = await request.json();
+
+  const result = await streamText({
+    model: openai('gpt-4o'),
+    system: 'You are a friendly assistant!',
+    messages: convertToCoreMessages(messages),
```
Comment (jeremyphilemon) [inline content/docs/04-ai-sdk-ui/04-generative-ui.mdx:126]: add `maxSteps`

File: content/docs/04-ai-sdk-ui/04-generative-ui.mdx (hunk 4 above)
Comment (nicoalbanese) [reply content/docs/04-ai-sdk-ui/04-generative-ui.mdx:20]: maybe we just remove that last bullet then?
